src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1478,12 +1478,15 @@ unsigned uct_dc_mlx5_ep_fc_hard_req_progress(void *arg)
     uint64_t ep_key;
     uct_dc_mlx5_ep_t *ep;
 
-    if (ucs_likely(now - iface->tx.fc_hard_req_resend_time)) {
+    if (ucs_likely(now < iface->tx.fc_hard_req_resend_time)) {
         return 0;
     }
 
+    /* Go over all endpoints which are waiting for FC window being restored and
+     * and resend FC_HARD_REQ packet to make sure a peer will resend
+     * FC_PURE_GRANT packet in case of failure */
```

Comment:
FC_PURE_GRANT packet in case of failure on the remote FC endpoint

Reply:
fixed